docs/dev/managing-dependencies.md

```diff
@@ -0,0 +1,105 @@
+# Managing dependencies in Gaffer
+
+Gaffer is a large project with a lot of dependency and transitive dependency management required. This page covers some ways to get information about and more easily visualise and understand the dependencies for given Maven modules (or even the whole project).
+
+For more information on how Maven handles dependencies, [see this Maven guide](https://maven.apache.org/guides/introduction/introduction-to-dependency-mechanism.html).
+
+
+## Get a simple list of direct dependencies
+
+This outputs the direct (non-transitive) dependencies for all Maven modules in the project. For specific modules append `-pl :module-name` to the command.
+```
+mvn dependency:list -DexcludeTransitive=true -Dsort=true
+```
+If `-DexcludeTransitive=true` is not used it will also print transitive dependencies, but without showing where they come from. The tree option (explained below) is better for getting this kind of information.
+
+For more info, [see the Maven dependency plugin documentation for the 'list' goal](https://maven.apache.org/plugins/maven-dependency-plugin/list-mojo.html).
+
+### Output to File
+
+The user property `-DoutputFile=deps.txt` can be used to place output for each module into the root directory **of that module**. It is also possible to aggregate all results into a single file using `mvn dependency:list -B -DappendOutput=true -DoutputFile=$(pwd)/mvn-dep-list.txt`.
+
+However, any files created contain terminal escape characters for adding colour (even when using `-B` - this is likely a bug). Another option is to use `-B` to turn off colour, piping the output through `sort --unique` (which also aggregates the dependencies) and appending that to a file - instead of using the plugin's file output option.
+
+
+## Get a tree showing Dependencies
+
+This produces a tree showing the **resolved** dependencies for all Maven modules in the project. For specific modules append `-pl :module-name` to the command (as before).
+```
+mvn dependency:tree
+```
+
+To include **all** dependencies (resolved, transitive and duplicate) in the tree and to see information on versions which have been managed (via `dependencyManagement`), append `-Dverbose`.
+
+This verbose flag is very useful for finding out which dependencies are affected by version management and version conflicts.
+
+Here's a simplified tree output with the verbose output explained (expand the :material-plus-circle-outline: symbols):
+```c
```

Comment:
Would bash work? Otherwise markdown is being told this is c code right?

Reply:
Bash looks wrong as it highlights things which have no significance here. What I'd like to do it have no highlighting at all. However, in that case there's no way for MkDocs to tell what's a comment and therefore the expanding 'plus' annotation doesn't work.

I chose C because it was the least worst option in terms of incorrect syntax highlighting while having the annotation still working.